File: ModuloJava/tcc/backend/redesocial/src/main/java/br/com/cwi/redesocial/service/cliente/LoginService.java
package br.com.cwi.redesocial.service.cliente;

import br.com.cwi.redesocial.dominio.Usuario;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCrypt;
import org.springframework.stereotype.Repository;

@Repository
public class LoginService {

    @Autowired
    private BuscaUsuarioPorEmailService buscaUsuarioPorEmailService;

    public String login(String email,String senha){
        Usuario usuarioCarregado = buscaUsuarioPorEmailService.buscar(email);

        if(!BCrypt.checkpw(senha,usuarioCarregado.getSenha())){
            throw new IllegalArgumentException("Usuario e senha invalido");
        }

        return usuarioCarregado.getSenha();
    }
}

File: ModuloJava/tcc/frontend/src/service/LoginService.js
import CONFIG from "../config";

import axios from "axios";

export default class LoginService{

    static login(email,password){
        var data = {
            email:email,
            senha:password,
        }
        return axios.post(`${CONFIG.API_URL_BASE}/public/usuario/login`,data)
    }

    static register(data){
        return axios.post(`${CONFIG.API_URL_BASE}/public/usuario`,data)
    }
}